extensions/ql-vscode/src/databases/config/db-config-store.ts

```diff
@@ -76,6 +78,84 @@ export class DbConfigStore extends DisposableObject {
     await this.writeConfig(config);
   }
 
+  public async removeDbItem(dbItem: DbItem): Promise<void> {
+    if (!this.config) {
+      throw Error("Cannot remove item if config is not loaded");
+    }
+
+    const config: DbConfig = cloneDbConfig(this.config);
+    const selectedItem: SelectedDbItem | undefined = config.selected;
+
+    // Remove item from databases
+    switch (dbItem.kind) {
+      case DbItemKind.LocalList:
+        config.databases.local.lists = config.databases.local.lists.filter(
+          (list) => list.name !== dbItem.listName,
+        );
+        break;
+      case DbItemKind.RemoteUserDefinedList:
+        config.databases.remote.repositoryLists =
+          config.databases.remote.repositoryLists.filter(
+            (list) => list.name !== dbItem.listName,
+          );
+        break;
+      case DbItemKind.LocalDatabase:
```

Comment:
Once we properly support local databases, will we need to think about removing DBs on disk as well? If so, maybe add a comment about that 🤔 (This is very much a future concern though, so feel free to ignore!)